File: app/src/main/java/com/qhy040404/libraryonetap/datamodel/VolunteerData.java
package com.qhy040404.libraryonetap.datamodel;

import com.google.gson.Gson;

import java.io.Serializable;

public class VolunteerData implements Serializable {
    Gson gson = new Gson();

    private static class GsonData {
        private int numSameID;
        private int numSameName;
        private double totalDuration;
    }

    public int getSameID(String data) {
        return gson.fromJson(data, GsonData.class).numSameID;
    }

    public int getSameName(String data) {
        return gson.fromJson(data, GsonData.class).numSameName;
    }

    public double getTotalHours(String data) {
        return gson.fromJson(data, GsonData.class).totalDuration;
    }
}

File: app/src/main/java/com/qhy040404/libraryonetap/datamodel/VolunteerData.kt
package com.qhy040404.libraryonetap.datamodel

import com.qhy040404.libraryonetap.constant.GlobalManager.gson

object VolunteerData {
    fun getSameID(data: String?): Int {
        return gson.fromJson(data, VolunteerDataClass::class.java).numSameID
    }

    fun getSameName(data: String?): Int {
        return gson.fromJson(data, VolunteerDataClass::class.java).numSameName
    }

    fun getTotalHours(data: String?): Double {
        return gson.fromJson(data, VolunteerDataClass::class.java).totalDuration
    }
}